The Economic Times.js

```diff
@@ -0,0 +1,176 @@
+{
+	"translatorID": "1a9a7ecf-01e9-4d5d-aa19-a7aa4010da83",
+	"label": "The Economic Times",
+	"creator": "Sonali Gupta",
+	"target": "http://economictimes.indiatimes.com",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2017-03-19 08:42:35"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2017 Sonali Gupta
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.indexOf("/topic/") != -1) {
+		return "multiple";
+	}
+	else if (ZU.xpathText(doc, '//article')) {
+		return "newspaperArticle";
+	}
+}
+
+function doWeb(doc, url) {
+
+	//srcaping titles of search results
+	if (detectWeb(doc, url) == "multiple") {
+		var myXPath = '//main/descendant::h3';
+		var myXPathObject = ZU.xpathText(doc, '//main/div[1]/div[2]/a/h2',',','*');
+		myXPathObject = myXPathObject.concat(ZU.xpathText(doc, myXPath,',','*'));
+		var array = myXPathObject.split('*')
+		var items = new Object();
+		var articles = new Array();
+		for (var i in array){
+			items[i]=array[i];
+		}
+		Zotero.selectItems(items, function (items) {
+			if (!items) {
+				return true;
+			}
+			for (var i in items) {
+				articles.push(i);
+			}
+			Zotero.Utilities.processDocuments(articles, scrape);
+		});
+	}	
+	else{
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
+	
+	newItem = new Zotero.Item("newspaperArticle");
+	newItem.url = url;
+	newItem.publicationTitle = "The Economic Times";
+
+	//get headline
+	var title = ZU.xpathText(doc, '//article/h1');
+	if (!title) title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	newItem.title = title;
+
+	//get abstract
+	var abstract = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
+	newItem.abstractNote = abstract;
+	
+	//get date and day
+	var date = ZU.xpathText(doc, '//article/div[4]/div[1]');
+	if(!date) date = ZU.xpathText(doc, '//article/div[1]/div[2]');
+	date = date.substring(date.indexOf("|") + 1);
+	var date_day = date.replace('Updated:','');
+	newItem.date = date_day;
```

Comment:
This looks very fragile, because it depends on the exact order of the nodes. Usually it is better to use some attributes to filter on the correct elements (especially `class`). Moreover, if the variables like `date` can still be `null`, we should test that before continue.

I suggest here something like:
```js
	var date = ZU.xpathText(doc, '//article/div/div[contains(@class, "publish_on") or contains(@class, "byline")]');
	if (date) {
		newItem.date = ZU.strToISO(date);
	}
```
Note: The `ZU.strToISO` is deleting most of the other noise around the date and it seems already to work good for a few examples I tested.